### ISSUE TITLE
upload more imformation when detecting OOS

### ISSUE THREAD

**Issue by gfgtdf** [wesnoth/wesnoth#4845]
There was recently a post in the forum that shared mp server analysis, and (at least for me) the most important oberveation was the fact that only a small fraction of OOS errors, is caused by bad wml in one of the active add-ons, so i think it'd be nice to have more information that help to investigateabout the casue of OOS erros.  in particular:

1) Whci other addons are installed adn their versions,( i think it' be nice to ask for confirmation from the users for this information. though)
2) checksums of the mainline config files and for each addon.
3) information about reassigning sides (using `:control` this is laos a common OOS source)

**Comment by Pentarctagon** [comment]
For (1), would the need for that would be removed if #4832 were implemented?
For (2), do you mean the checksum for each individual mainline cfg file?  Also, what about lua files?
For (3), I'd guess that would be fairly easy to do - maybe just add an insert to a new `game_commands` table.

**Comment by gfgtdf** [comment]
> For (1), would the need for that would be removed if #4832 were implemented

Yes but this fe is also about 1.14 and #4832 most likely won't go I to 1.14

> For (2), do you mean the checksum for each individual mainline cfg file

I kinda thought of a checksum of the whole data/ directory,  and one for each addon directory. I'm filled aware that some things can be changed without causing OOS, so a different checksum doesn't imply oos, I still think this is useful to have


> For (3), I'd guess that would be fairly easy to do - maybe just add an insert to a new game_commands table.


I'm in particular interested in the case where the host removes control from a side and then gives that side control back without him leaving+rejoining first

**Comment by Pentarctagon** [comment]
So (3) is the use of `:control`/`give_control` as well as players joining/leaving?

**Comment by gfgtdf** [comment]
> So (3) is the use of :control/:give_control as well as players joining/leaving?

No, what i was saying is that if you remove control from a player via :control, then that players client  often goes into a corrupt state, which causes OOS, This is in particular a problem if that player regains control of a side later (since OOS aren't that bad if there are just on a oververs client)

**Comment by Pentarctagon** [comment]
We'd need to know when the leaving/joining happened to be able to tell if a player had their control of a side removed and then given back without the player leaving in between though.

**Comment by Wedge009** [comment]
This is the last issue outstanding that's tagged specifically for 1.14 and there are no more releases planned for that branch. Is this really specific to 1.14 or can/does it apply to ongoing development as well?

**Comment by ProditorMagnus** [comment]
Not specific.
